fud/fud/config.py

```diff
@@ -285,6 +289,62 @@ def setup_external_stage(self, args):
             else:
                 log.error(f"No external script named `{args.name}'.")
 
+    def discover_implied_states(self, filename):
+        """
+        Use the mapping from filename extensions to stages to figure out which
+        states were implied.
+        Returns the input state on which the implied stage operates
+        """
+        suffix = Path(filename).suffix
+        stages = []
+        for (name, stage) in self["stages"].items():
+            if "file_extensions" in stage:
+                for ext in stage["file_extensions"]:
+                    if suffix == ext:
+                        stages.append(name)
```

Comment:
```suggestion
        for (name, stage) in self["stages"].items():
            if "file_extensions" not in stage:
                continue
            stages.extend(name if ext == suffix for ext in stage["file_extensions"])
```

Reply:
Feel free to ignore, just personal preference of avoiding pyramids. Is this intended behavior as well? This could append the same name multiple times right?

Reply:
Looks reasonable!